cobalt/build/test_targets.py

```diff
@@ -0,0 +1,259 @@
+    '//ipc:ipc_perftests',
+    '//media/cast',
+    '//mojo/core:mojo_core_unittests',
+    '//mojo/core:mojo_perftests',
+    '//net:net_perftests',
+    '//pdf',
+    '//testing/libfuzzer/tests:libfuzzer_tests',
+    '//third_party/angle',
+    '//third_party/dawn',
+    '//third_party/pdfium',
+    '//third_party/libaddressinput',
+    '//third_party/libphonenumber',
+    '//third_party/swiftshader',
+    '//ui/shell_dialogs',
+    '//weblayer',
+}
+
+# The two nodes are adjacent in a path that contains many unrelated components.
+# Targets in these paths must share an additional path segment to be considered
+# related and subject for test target considerations.
+_UNRELATED_TARGETS_PATHS = {'components', 'third_party', 'ui'}
+
+
+def _are_related(node1, node2) -> bool:
+  """Two targets are considered related if the share at least one path segment
+  or two or more if they are in one of the paths that contain many unrelated
+  targets.
+  """
+  # Remove the target name and split the path into its segments.
+  node1_path = node1.split(':')[0].split('/')
+  node2_path = node2.split(':')[0].split('/')
+  common = os.path.commonprefix([node1_path, node2_path])
+  if len(common) == 2:
+    # Only the initial // matches.
+    return False
+  # At least one path segment matches.
+  return len(common) > 3 or common[2] not in _UNRELATED_TARGETS_PATHS
+
+
+def _is_test_target(g, node) -> bool:
+  if not node in _ALLOW_TESTS or \
+    any(node.startswith(path_prefix) for path_prefix in _EXCLUDE_TESTS):
+    return False
+
+  # Test targets get a runner target added to its deps. On linux the name of
+  # this target is ${target_name}__runner, on android it's
+  # ${target_name}__test_runner_script.
+  # See src/testing/test.gni for the test() template definition.
+  linux_runner = f'{node}__runner'
+  android_runner = f'{node}__test_runner_script'
+  successors = set(g.successors(node))
+  return linux_runner in successors or android_runner in successors
+
+
+def _create_graph(data: dict) -> Tuple[nx.DiGraph, Dict[str, List[str]]]:
+  """Creates a directed graph from 'gn desc' json output.
+  Using targets as nodes and creating edges from dependencies.
+  """
+  g = nx.DiGraph()
+  source_map = collections.defaultdict(list)
+  for target_name, attributes in data.items():
+    node_attributes = {
+        'outputs': attributes.get('outputs', []),
+        'type': attributes['type'],
+    }
+    g.add_node(target_name, **node_attributes)
+    g.add_edges_from((target_name, dep) for dep in attributes['deps'])
+    for source_file in attributes.get('inputs', []):
+      source_map[source_file] += [target_name]
+    for source_file in attributes.get('sources', []):
+      source_map[source_file] += [target_name]
+    source_map[target_name.split(':')[0]] += [target_name]
+  return g, source_map
+
+
+def _add_test_target(g: nx.DiGraph, target: str, test_targets: Set[str],
+                     executables: Set[str]):
+  test_targets.add(target[2:])
```

Comment:
Using a tiny helper function to strip out the leading `//` or using `lstrip('/')` would be easier to understand here